File: shared/src/commonMain/kotlin/dem/dev/timeflame/util/state/ScreenState.kt
package dem.dev.timeflame.util.state

enum class ResultType {
    SUCCESS, FAILURE
}

data class Loading(val messageCode: Int? = null, val silentMode: Boolean = false /* Silent mode: don't show loading dialog */)
data class Result(val resultType: ResultType, val messageCode: Int? = null)

sealed interface ScreenState {
    data class Loading(val messageCode: Int? = null, val silentMode: Boolean = false /* Silent mode: don't show loading dialog */): ScreenState
    data class Result(val resultType: ResultType, val messageCode: Int? = null): ScreenState
    data object Idle: ScreenState
}

File: iosApp/iosApp/core/util/ScreenState.swift
//
//  ScreenState.swift
//  iosApp
//
//  Created by Mikhail Maevskii on 30/10/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation

enum ScreenState {
    case idle
    case loading(messageCode: Int)
    case result(resultType: ResultType, messageCode: Int)
}
